bridge-sdk/connectors/fast-bridge/src/fast_bridge.rs

```diff
@@ -0,0 +1,355 @@
+        fee_amount: u128,
+        eth_token_address: Address,
+        recipient: Address,
+        valid_till: u64,
+    ) -> Result<CryptoHash> {
+        let near_endpoint = self.near_endpoint()?;
+        let fast_bridge_account_id = self.fast_bridge_account_id()?.to_string();
+
+        let message = TransferMessage {
+            valid_till,
+            transfer: TransferDataEthereum {
+                token_near: token_id.clone(),
+                token_eth: EthAddress(eth_token_address.0),
+                amount,
+            },
+            fee: TransferDataNear {
+                token: token_id.clone(),
+                amount: fee_amount,
+            },
+            recipient: EthAddress(recipient.0),
+            valid_till_block_height: None,
+            aurora_sender: None,
+        };
+
+        let mut buffer: Vec<u8> = Vec::new();
+        message
+            .serialize(&mut buffer)
+            .map_err(|_| BridgeSdkError::UnknownError)?;
+        let msg = BASE64_STANDARD.encode(&buffer);
+
+        let args = format!(
+            r#"{{"receiver_id":"{fast_bridge_account_id}","amount":"{amount}","msg":"{msg}"}}"#
+        )
+        .to_string()
+        .into_bytes();
+
+        let tx_hash = near_rpc_client::change(
+            near_endpoint,
+            self.near_signer()?,
+            token_id.to_string(),
+            "ft_transfer_call".to_string(),
+            args,
+            200_000_000_000_000,
+            1,
+        )
+        .await?;
+
+        tracing::info!(
+            tx_hash = format!("{:?}", tx_hash),
+            "Sent tokens to the fast bridge contract"
+        );
+
+        Ok(tx_hash)
+    }
+
+    /// Completes fast bridge transfer by sending tokens to the recipient on Ethereum. The proof from this transaction is to be used to unlock tokens on NEAR for unlock_recipient
+    #[tracing::instrument(skip_all, name = "TRANSFER ON ETH")]
+    pub async fn transfer_on_eth(
+        &self,
+        token: Address,
+        recipient: Address,
+        nonce: U256,
+        amount: U256,
+        unlock_recipient: String,
+        valid_till_block_height: U256,
+    ) -> Result<TxHash> {
+        let fast_bridge = self.fast_bridge_contract()?;
+        let transfer_call = fast_bridge
+            .transfer_tokens(
+                token,
+                recipient,
+                nonce,
+                amount,
+                unlock_recipient,
+                valid_till_block_height,
+            )
+            .value(amount);
+
+        let tx = transfer_call.send().await?;
+
+        tracing::info!(
+            tx_hash = format!("{:?}", tx.tx_hash()),
+            "Completed fast bridge transfer"
+        );
+
+        Ok(tx.tx_hash())
+    }
+
+    /// Unlocks tokens on Near following a successful transfer completion on Ethereum.
+    #[tracing::instrument(skip_all, name = "LP UNLOCK")]
+    pub async fn lp_unlock(&self, tx_hash: TxHash, log_index: u64) -> Result<CryptoHash> {
```

Comment:
I need instructions on how to find the correct log index.

Reply:
In the future we can find the index ourselves. For now it can be found on the logs tab in etherscan https://sepolia.etherscan.io/tx/0x6b8712d6bf767dfd37a4ff78dd84b5f01b3b8cab8d4f831d90ab6c29d6d8fb16#eventlog

Reply:
https://github.com/Near-One/rainbow-bridge-sdk-rs/issues/11